### PR TITLE
Fix #1119: Lowfi Recently Played

### LINKED ISSUE #1119: Lowfi: Recently Played
Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/2b6c8e95-5ac7-4472-bc2b-1c318a3f7e5c/HP-Continue-Playing-

Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0e6ba726-1930-4f39-9c73-51df10a1c766/HP-Continue-Playing-

As per the new mocks, the layout arrangement of items has changed and from each item the "Chapter name" has been removed.

This is issue is tracking the new low-fi implementation for RecentlyPlayed.

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

Fixes #1119 

Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/2b6c8e95-5ac7-4472-bc2b-1c318a3f7e5c/HP-Continue-Playing-

Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0e6ba726-1930-4f39-9c73-51df10a1c766/HP-Continue-Playing-

As per the new mocks, the layout arrangement of items has changed and from each item the "Chapter name" has been removed.

All test cases are passing Espresso Pixel Device.

Screenshot_1590206828
Screenshot_1590206836
Screenshot_1590206864
Screenshot_1590206881
Screenshot_1590207312
Screenshot_1590207315

## How to test
Refernece: https://gist.github.com/rt4914/5258375c8d2cca81b6a5b62ccab8e143#file-homefragmentpresenter-kt-L92
1. To add items for recent stories:
```
storyProgressTestHelper.markRecentlyPlayedForFractionsStory0Exploration0(profileId, timestampOlderThanAWeek = false)
```

2. To add items for old stories:
```
 storyProgressTestHelper.markRecentlyPlayedForRatiosStory0Exploration0(profileId, timestampOlderThanAWeek = true)
```

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/home/recentlyplayed/OngoingListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package org.oppia.app.home.recentlyplayed
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.oppia.app.databinding.OngoingStoryCardBinding
+import org.oppia.app.databinding.SectionTitleBinding
+import org.oppia.app.home.recentlyplayed.OngoingStoryViewModel
+import org.oppia.app.home.recentlyplayed.RecentlyPlayedItemViewModel
+import org.oppia.app.home.recentlyplayed.SectionTitleViewModel
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/home/recentlyplayed/OngoingListAdapter.kt]: Remove unused imports

File: app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedActivity.kt [hunk 2]
```diff
@@ -42,4 +42,4 @@ class RecentlyPlayedActivity : InjectableAppCompatActivity(), RouteToExploration
       )
     )
   }
-}
+}
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedActivity.kt]: No EOF found

File: app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedFragment.kt [hunk 3]
```diff
@@ -41,4 +41,4 @@ class RecentlyPlayedFragment : InjectableFragment(), OngoingStoryClickListener {
   override fun onOngoingStoryClicked(promotedStory: PromotedStory) {
     recentlyPlayedFragmentPresenter.onOngoingStoryClicked(promotedStory)
   }
-}
+}
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedFragment.kt]: No EOF found

**Review by MohamedMedhat1998 — CHANGES_REQUESTED** [review]
3 Test cases are failing but I'm not sure whether the problem is with my emulator or with something else.  
Emulator:  
```
Nexus S API 22 x86
4.0 480x800 hdpi
Graphics: Hardware - GLES 2.0
Multi-core CPU: 2
RAM: 512 MB
VM Heap: 512 MB
```
Test cases that fail:  
1. `testRecentlyPlayedTestActivity_recyclerViewItem1_lessonThumbnailIsCorrect()` 
```

androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'with drawable from resource id: <2131230940>' doesn't match the selected view.
Expected: with drawable from resource id: <2131230940>[lesson_thumbnail_graphic_duck_and_chicken]
Got: "AppCompatImageView{id=2131296589, res-name=lesson_thumbnail, desc=Image for What is a Fraction?, visibility=VISIBLE, width=216, height=162, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@37d61412, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"

at dalvik.system.VMStack.getThreadStackTrace(Native Method)
at java.lang.Thread.getStackTrace(Thread.java:580)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:96)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:59)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:324)
at androidx.test.espresso.ViewInteraction.check(ViewInteraction.java:306)
at org.oppia.app.home.RecentlyPlayedFragmentTest.testRecentlyPlayedTestActivity_recyclerViewItem1_lessonThumbnailIsCorrect(RecentlyPlayedFragmentTest.kt:254)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:104)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1853)
Caused by: junit.framework.AssertionFailedError: 'with drawable from resource id: <2131230940>' doesn't match the selected view.
Expected: with drawable from resource id: <2131230940>[lesson_thumbnail_graphic_duck_and_chicken]
Got: "AppCompatImageView{id=2131296589, res-name=lesson_thumbnail, desc=Image for What is a Fraction?, visibility=VISIBLE, width=216, height=162, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@37d61412, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"

at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:540)
at androidx.test.espresso.assertion.ViewAssertions$MatchesViewAssertion.check(ViewAssertions.java:105)
at androidx.test.espresso.ViewInteraction$SingleExecutionViewAssertion.check(ViewInteraction.java:425)
at androidx.test.espresso.ViewInteraction$2.call(ViewInteraction.java:288)
at androidx.test.espresso.ViewInteraction$2.call(ViewInteraction.java:272)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:135)
at android.app.ActivityThread.main(ActivityThread.java:5254)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```
2. `testRecentlyPlayedTestActivity_changeConfiguration_recyclerViewItem1_lessonThumbnailIsCorrect()`
```

androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'with drawable from resource id: <2131230940>' doesn't match the selected view.
Expected: with drawable from resource id: <2131230940>[lesson_thumbnail_graphic_duck_and_chicken]
Got: "AppCompatImageView{id=2131296589, res-name=lesson_thumbnail, desc=Image for What is a Fraction?, visibility=VISIBLE, width=243, height=182, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@f739d39, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"

at dalvik.system.VMStack.getThreadStackTrace(Native Method)
at java.lang.Thread.getStackTrace(Thread.java:580)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:96)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:59)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:324)
at androidx.test.espresso.ViewInteraction.check(ViewInteraction.java:306)
at org.oppia.app.home.RecentlyPlayedFragmentTest.testRecentlyPlayedTestActivity_changeConfiguration_recyclerViewItem1_lessonThumbnailIsCorrect(RecentlyPlayedFragmentTest.kt:462)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:104)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1853)
Caused by: junit.framework.AssertionFailedError: 'with drawable from resource id: <2131230940>' doesn't match the selected view.
Expected: with drawable from resource id: <2131230940>[lesson_thumbnail_graphic_duck_and_chicken]
Got: "AppCompatImageView{id=2131296589, res-name=lesson_thumbnail, desc=Image for What is a Fraction?, visibility=VISIBLE, width=243, height=182, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@f739d39, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"

at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:540)
at androidx.test.espresso.assertion.ViewAssertions$MatchesViewAssertion.check(ViewAssertions.java:105)
at androidx.test.espresso.ViewInteraction$SingleExecutionViewAssertion.check(ViewInteraction.java:425)
at androidx.test.espresso.ViewInteraction$2.call(ViewInteraction.java:288)
at androidx.test.espresso.ViewInteraction$2.call(ViewInteraction.java:272)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:135)
at android.app.ActivityThread.main(ActivityThread.java:5254)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```
3. `testRecentlyPlayedTestActivity_recyclerViewItem1_clickStory_intentsToExplorationActivity()`
```

androidx.test.espresso.PerformException: Error performing 'single click - At Coordinates: 119, 311 and precision: 16, 16' on view 'Animations or transitions are enabled on the target device.
For more info check: https://developer.android.com/training/testing/espresso/setup#set-up-environment

with id: org.oppia.app:id/ongoing_story_recycler_view'.
at androidx.test.espresso.PerformException$Builder.build(PerformException.java:86)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:87)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:59)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:322)
at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:178)
at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:119)
at org.oppia.app.home.RecentlyPlayedFragmentTest.testRecentlyPlayedTestActivity_recyclerViewItem1_clickStory_intentsToExplorationActivity(RecentlyPlayedFragmentTest.kt:275)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:104)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1853)
Caused by: java.lang.RuntimeException: PARAGRAPH span must end at paragraph boundary
at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:629)
at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:607)
at android.text.HtmlToSpannedConverter.convert(Html.java:468)
at android.text.Html.fromHtml(Html.java:136)
at androidx.core.text.HtmlCompat.fromHtml(HtmlCompat.java:165)
at org.oppia.util.parser.HtmlParser.parseOppiaHtml(HtmlParser.kt:50)
at org.oppia.app.player.state.StateFragmentPresenter$createRecyclerViewAdapter$18.invoke(StateFragmentPresenter.kt:244)
at org.oppia.app.player.state.StateFragmentPresenter$createRecyclerViewAdapter$18.invoke(StateFragmentPresenter.kt:100)
at org.oppia.app.recyclerview.BindableAdapter$MultiTypeBuilder$registerViewBinder$viewHolderFactory$1$1.bind$app_debug(BindableAdapter.kt:229)
at org.oppia.app.recyclerview.BindableAdapter.onBindViewHolder(BindableAdapter.kt:84)
at org.oppia.app.recyclerview.BindableAdapter.onBindViewHolder(BindableAdapter.kt:28)
at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7065)
at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7107)
at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6012)
at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6279)
at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6118)
at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6114)
at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2303)
at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1627)
at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4134)
at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:3851)
at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4404)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:579)
at android.widget.FrameLayout.onLayout(FrameLayout.java:514)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:579)
at android.widget.FrameLayout.onLayout(FrameLayout.java:514)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at androidx.constraintlayout.widget.ConstraintLayout.onLayout(ConstraintLayout.java:1915)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:579)
at android.widget.FrameLayout.onLayout(FrameLayout.java:514)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:579)
at android.widget.FrameLayout.onLayout(FrameLayout.java:514)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1703)
at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1557)
at android.widget.LinearLayout.onLayout(LinearLayout.java:1466)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:579)
at android.widget.FrameLayout.onLayout(FrameLayout.java:514)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1703)
at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1557)
at android.widget.LinearLayout.onLayout(LinearLayout.java:1466)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:579)
at android.widget.FrameLayout.onLayout(FrameLayout.java:514)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1703)
at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1557)
at android.widget.LinearLayout.onLayout(LinearLayout.java:1466)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:579)
at android.widget.FrameLayout.onLayout(FrameLayout.java:514)
at android.view.View.layout(View.java:15671)
at android.view.ViewGroup.layout(ViewGroup.java:5038)
at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2086)
at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1843)
at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1061)
at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5885)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:767)
at android.view.Choreographer.doCallbacks(Choreographer.java:580)
at android.view.Choreographer.doFrame(Choreographer.java:550)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at androidx.test.espresso.base.Interrogator.loopAndInterrogate(Interrogator.java:148)
at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:525)
at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:484)
at androidx.test.espresso.base.UiControllerImpl.injectMotionEvent(UiControllerImpl.java:236)
at androidx.test.espresso.action.MotionEvents.sendUp(MotionEvents.java:162)
at androidx.test.espresso.action.MotionEvents.sendUp(MotionEvents.java:139)
at androidx.test.espresso.action.Tap.sendSingleTap(Tap.java:170)
at androidx.test.espresso.action.Tap.access$100(Tap.java:31)
at androidx.test.espresso.action.Tap$1.sendTap(Tap.java:47)
at androidx.test.espresso.action.GeneralClickAction.perform(GeneralClickAction.java:137)
at androidx.test.espresso.ViewInteraction$SingleExecutionViewAction.perform(ViewInteraction.java:366)
at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:255)
at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:65)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:158)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:155)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:135)
at android.app.ActivityThread.main(ActivityThread.java:5254)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

File: app/src/main/java/org/oppia/app/home/recentlyplayed/OngoingListAdapter.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/home/recentlyplayed/OngoingListAdapter.kt]: Done.

File: app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedActivity.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedActivity.kt]: Done.

File: app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedFragment.kt (hunk 3 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/home/recentlyplayed/RecentlyPlayedFragment.kt]: Done.

**Review by rt4914 — COMMENTED** [review]
@MohamedMedhat1998 Regarding test cases.
Actually we are suppose to check test cases on Pixel devices.

Also, for this specific device I checked on API 25 (NEXUS S) and it worked correctly. Unfortunately I am not able to run app on `Neuxs S API 22` as I am getting error in that and the solutions mentioned on internet are not working.

Screenshot 2020-05-26 at 12 00 07 PM

File: app/src/sharedTest/java/org/oppia/app/home/RecentlyPlayedFragmentTest.kt [hunk 4]
```diff
@@ -499,5 +773,20 @@ class RecentlyPlayedFragmentTest {
       throw UnsupportedOperationException()
     }
   }
-}
 
+  /** Custom class to check number of spans occupied by an item at a given position. */
```
Comment (veena14cs) [inline app/src/sharedTest/java/org/oppia/app/home/RecentlyPlayedFragmentTest.kt]: Can we have this class in Utility folder of androidTest.

**Review by veena14cs — APPROVED** [review]
Please make one change otherwise good to merge.

File: app/src/sharedTest/java/org/oppia/app/home/RecentlyPlayedFragmentTest.kt (hunk 4 above)
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/home/RecentlyPlayedFragmentTest.kt]: Shifted this class to `RecyclerViewMatcher` along with its corresponding function and used it in this file.

Comment (veena14cs) [reply app/src/sharedTest/java/org/oppia/app/home/RecentlyPlayedFragmentTest.kt]: LGTM.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM, besides, test-cases are failing in lower API as already mentioned in the PR and also failing while running with Robolectric configuration changes.

**Review by MohamedMedhat1998 — APPROVED** [review]
LGTM

**Comment by rt4914** [comment]
@MohamedMedhat1998 Please do not merge PR by other authors without confirming with them. If you were blocked because of you this PR, then in that case you can easily checkout from this branch and create your branch on top of this one, that way you won't.

**Comment by rt4914** [comment]
> LGTM, besides, test-cases are failing in lower API as already mentioned in the PR and also failing while running with Robolectric configuration changes.

Sounds good. This is getting tracked in https://github.com/oppia/oppia-android/issues/973 so I will take care of it. Thanks.

**Pull request merged**